lib/widgets/action_sheet.dart

```diff
@@ -1,36 +1,176 @@
 import 'package:flutter/material.dart';
 import 'package:share_plus/share_plus.dart';
 
+import '../api/exception.dart';
 import '../api/model/model.dart';
+import '../api/route/messages.dart';
+import 'compose_box.dart';
+import 'dialog.dart';
 import 'draggable_scrollable_modal_bottom_sheet.dart';
+import 'message_list.dart';
+import 'store.dart';
 
+/// Show a sheet of actions you can take on a message in the message list.
+///
+/// Must have a [MessageListPage] ancestor.
 void showMessageActionSheet({required BuildContext context, required Message message}) {
+  // The UI that's conditioned on this won't live-update during this appearance
+  // of the action sheet (we avoid calling composeBoxControllerOf in a build
+  // method; see its doc). But currently it will be constant through the life of
+  // any message list, so that's fine.
+  final isComposeBoxOffered = MessageListPage.composeBoxControllerOf(context) != null;
   showDraggableScrollableModalBottomSheet(
     context: context,
-    builder: (BuildContext context) {
+    builder: (BuildContext innerContext) {
```

Comment:
That's true as of this later commit:

```
action_sheet [nfc]: Pull out base class for message action sheet buttons
```

so I'll plan to make this change there, instead of here in this commit.

Reply:
Oh, I think maybe you put this comment on that commit I referred to, and GitHub just doesn't make it obvious whether it was meant for that commit or

```
action_sheet [nfc]: Rename a param to stop shadowing
```

or even some other commit.

Reply:
> Oh, I think maybe you put this comment on that commit I referred to, and GitHub just doesn't make it obvious whether it was meant for that commit or

As far as GitHub sees, I put the comment on the PR's overall diff. My usual review workflow is that I read the changes commit by commit, in `git log --stat -p --reverse`, but make comments all on the one main page of the "Files changed" tab on the PR in GitHub.

When I made the comment, I was looking at the "Rename a param to stop shadowing" commit, but also consulted the branch tip's version in the IDE to confirm what the state of things was at the end.

I think possibly an ideal version of this branch would have that "Rename a param to stop shadowing" commit happen later after that "Pull out base class" commit, or not at all. But wherever in the branch you find to cleanly make the change is fine.